### DISCUSSION TITLE
Does/Will Appwrite Support Database Schemas/Models

### DISCUSSION THREAD

**Discussion by FXschwartz** [appwrite/appwrite#3704] [Q&A]
Similar to how [Cloud Firestore ODM](https://github.com/firebase/flutterfire/tree/master/packages/cloud_firestore_odm) helps with quering, filtering, and mapping database results over to local models/schemas.

Does Appwrite support this? Or if not is it something that's planned in the future?

With some added context I plan on using MongoDB when support for it is completed.

It's one of the few things keeping me from switching over!

**Comment by TorstenDittmann** [comment]
Hey 👋🏻 

We have already played with the idea, but that will be a bigger undertaking in terms of maintainability and initial work on the SDK's. Considering that we are striving to offer the same functionality on all SDK's. 

We are definitely planning to allow generating Typings/Models from your Database Collections, not sure if it will be part of the CLI or a separate package for each respective language.

Though it will not replace an ODM/ORM in any way, but a first good step in that direction.
